src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -310,22 +310,23 @@ public sealed class SecurityIdentifier : IdentityReference, IComparable<Security
         //       the typecast
         //
 
-        private void CreateFromParts(IdentifierAuthority identifierAuthority, int[] subAuthorities)
+        private void CreateFromParts(IdentifierAuthority identifierAuthority, ReadOnlySpan<int> subAuthorities)
         {
-            if (subAuthorities == null)
+            if (subAuthorities.IsEmpty)
             {
                 throw new ArgumentNullException(nameof(subAuthorities));
```

Comment:
Is this reachable?  Wondering if it should be an assert instead.

If it is reachable, is it possible this is now throwing for an empty input whereas previously it would only throw for a null input?

Reply:
This does seem like it would be better as `Debug.Assert(subAuthorities.Length > 0)`.  All current callers are passing values downstream of `new int[]`.  The only one that's possibly a concern is CreateFromBinaryForm, if it gets written to hybridize stack and array.  But that just means that CreateFromBinaryForm needs to handle the zero case itself.  (And probably needs a test that passes 0 sub-authorities, to make sure we don't regress behavior).

Reply:
The check is invalid, 0 is a valid and used length. The null check has been hoist out to the caller for the ctor and no other case can pass in null.